packages/knip/src/reporters/gitlab.ts

```diff
@@ -0,0 +1,128 @@
+import { createHash } from 'node:crypto';
+import type { Entries } from 'type-fest';
+import type { Issue, IssueRecords, IssueSeverity, IssueSymbol, Report, ReporterOptions } from '../types/issues.js';
+import { getTitle } from './util.js';
+
+/**
+ * Entry for GitLab Code Quality report. See: https://docs.gitlab.com/ci/testing/code_quality/#code-quality-report-format
+ */
+interface CodeQualityEntry {
+  check_name: string;
+  description: string;
+  severity: 'info' | 'minor' | 'major' | 'critical' | 'blocker';
+  fingerprint: string;
+  location: {
+    path: string;
+    lines: {
+      begin: number;
+    };
+  };
+}
```

Comment:
Looking at the specs, I would love to name it `codeclimate` while keeping it compatible for GitLab usage. Would that work for you as well?

If so, could you please add `"type": "issue"` to the interface/object?

I'm not sure about `categories` yet, so let's leave it out for now.

Reply:
I also add the categories array and use "Duplication" or "Bug Risk" for it.